Fix most golint warnings.

juliusv opened the pull request:
This is with `golint -min_confidence=0.5`.

I left 7 remaining lint warnings untouched because they were either
incorrect or I felt it was better not to change them at the moment.

juliusv commented:
@fabxc

fabxc commented on retrieval/discovery/kubernetes/types.go:
```diff
@@ -13,27 +13,27 @@
 
 package kubernetes
 
```
Please undo any changes to this file. It is essentially vendored from a Kubernetes package and modifying it will just cause more work if their API package becomes reasonable to use in general.

fabxc commented on promql/ast.go:
```diff
@@ -197,14 +213,18 @@ func (*StringLiteral) expr()  {}
+	// CardOneToMany represents one-to-many vector matching cardinality.
 	CardOneToMany
+	// CardManyToMany represents many-to-many vector matching cardinality.
 	CardManyToMany
```
Essentially all changes in this file are code-obfuscations to me. Them coming from a tool doesn't make them universally correct.
Especially this one – each comment is the type and constant name again and again.

fabxc commented on promql/ast.go:
```diff
@@ -171,15 +171,31 @@ type VectorSelector struct {
+// Type implements Node.
+func (e *StringLiteral) Type() model.ValueType { return model.ValString }
+
+// Type implements Node.
```
Also code obfuscation – those type of "implements" comments might make partially sense. In general they seem off because they turn Go's implicit interface back into Java style where a type has to specify what it implements.

We just went from a compact representation of which node type has which value type to the same thing, just messier and with a useless comment.

fabxc commented:
I know you feel stronger about complying with the tool, so :+1: 
Many fixes are obviously good.

juliusv commented:
Addressed all comments. Initially I wanted to fix *all* golint comments, but then realized that many are bogus - then I read the golint README after this PR and realized it's not meant to be ever used for presubmit-check style tests :) Too bad there isn't a linting/checking tool that makes sense to always follow.

fabxc commented:
\o/ :+1: :tada:
